File: problem-types/caskets.py
import sys

def entry_form(data, kwargs):
	names = {
	'here': 'Приз лежит здесь',
	'no': 'Здесь приза нет',
	'near': 'Приз в соседней шкатулке'
	}
	yield '<div class="row">'
	num = 1
	for name in data['tmp']:
		yield '<div class="column">'
		yield f'<img class="casket" tmp="1" num="{num}" src="/static/casket/empty_1.png" />'
		yield f'<div class="name"> {names[name]} </div>'
		yield '</div>'
		num += 1
	yield '</div>'

def steps(step_num, params, data):
	if step_num > 1:
		return {'answer': 'no'}
	print(data['correct'], params['answer'], file=sys.stderr)
	if int(data['correct']) == int(params['answer']):
		return {'answer': 'true', 'user_answer': params['answer'], 'answer_correct': True, 'solution': params['solution_full']}
	else:
		return {'answer': 'false', 'user_answer': params['answer'], 'answer_correct': False, 'solution': params['solution_empty']}

def validate(data, answer):
	return int(data['correct']) == int(answer)

HINT_ONLY = True

File: problem-types/caskets.js
function open_casket(casket, config) {
	if ($(casket).attr('tmp') >= 6) {
		clearInterval(movement)
		window.location.reload('true')
		return;
	}
	$(casket).attr('tmp', casket.getAttribute('tmp') - -1)
	$(casket).attr('src', '/static/problem_assets/casket/' + config + '_' + $(casket).attr('tmp') + '.png') 
}

function send_xhr(params) {
	let url = new URL(window.location.href + 'xhr')
	let casket = params['casket']
	$(casket).attr('src', '/static/problem_assets/casket/empty_6.png')
	let solution_empty = $('#problem_form')[0].outerHTML
	$(casket).attr('src', '/static/problem_assets/casket/full_6.png')
	let solution_full = $('#problem_form')[0].outerHTML
	$(casket).attr('src', '/static/problem_assets/casket/empty_1.png')
	$.post(url, JSON.stringify({'answer': casket.getAttribute('num'), 'solution_empty': solution_empty, 'solution_full': solution_full}), function(data){
		if (data == 'true')
			movement = setInterval(function(){open_casket(casket, 'full')}, 30)
		else if (data == 'false')
			movement = setInterval(function(){open_casket(casket, 'empty')}, 30)
		else 
			show_xhr('Попытки закончились!')
	})
}

var movement = ''
$(".casket").on("click touchstart", function(){confirm_action('Вы уверены, что хотите выбрать эту шкатулку? Больше попыток не будет.', 
	['Подтердить', 'Отмена'], 
	[send_xhr, function(){}],
	[{'casket': this}, {}])})